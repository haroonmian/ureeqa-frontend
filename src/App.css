.App{
    width: 100%;
    height:100vh;
    display: flex;
    flex-direction: column; 
    align-items: center;
    justify-content: center;
}
.valueShow{
letter-spacing: 3px;
}
#innum{
    margin: 10px;
}
.answers{
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-around;
    width: 30%;
}