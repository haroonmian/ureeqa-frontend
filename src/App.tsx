import React, { useState } from 'react';
import './App.css';
import axios from "axios";

function App() {
  const [dats, setdats] = useState<string>("hello")
  function getFetcher() {

    axios.defaults.baseURL = "http://localhost:4000/"
    const input = document.getElementById('innum') as HTMLInputElement | null;
console.log(input)
    axios.get(`/api/v1/prime?num=${input || input !=null ? input.value : 0}`)
      .then(function (response: any) {
        console.log(response.data.output);
        setdats(response.data.output)
      })
      .catch(function (error: any) {
        console.log(error);
      })
  }
  return (
    <>
      <div className="App">
        <h1>Get Your Prime Numbers</h1>
        {/* <label htmlFor="number">Number:</label> */}
        <input type="number" id="innum" name="number" data-testid="myinput"/>
        <button onClick={() => getFetcher()} >Get It</button>
        <h2 className="valueShow" id="result" data-testid="dats">{dats[0]}</h2>

      </div>
    </>
  );
}

export default App;
