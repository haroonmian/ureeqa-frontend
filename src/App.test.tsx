import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import App from './App';

  test("email input should accept text",async ()=>{
    render(<App/>)
    
    const emailInputNode = screen.getByTestId('myinput');
    const getButton=screen.getByText("Get It")
    const result = screen.getByTestId("dats")

      // expect(emailInputNode.value).toMatch("18")
    fireEvent.change(emailInputNode, {target: {value: '18'}} )
    fireEvent.click(getButton)
    await new Promise((r) => setTimeout(r, 2000));
    expect(result.innerHTML).toEqual("7")
  })
