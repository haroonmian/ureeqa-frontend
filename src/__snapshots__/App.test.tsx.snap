// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`email input should accept text 1`] = `
<h2
  class="valueShow"
  data-testid="dats"
>
  hello
</h2>
`;

exports[`email input should accept text: 
  <h2
  className="valueShow"
  data-testid="dats"
  >
  7
  </h2>
   1`] = `
<h2
  class="valueShow"
  data-testid="dats"
>
  hello
</h2>
`;

exports[`email input should accept text: 
  <h2
  className="valueShow"
  data-testid="dats"
  >
  10
  </h2>
   1`] = `
<h2
  class="valueShow"
  data-testid="dats"
>
  hello
</h2>
`;

exports[`email input should accept text: 
  <h2
  className="valueShow"
  data-testid="dats"
  >
  17
  </h2>
   1`] = `
<h2
  class="valueShow"
  data-testid="dats"
>
  hello
</h2>
`;

exports[`email input should accept text: 
  <h2 className="valueShow" data-testid="dats">{dats}</h2>
   1`] = `
<h2
  class="valueShow"
  data-testid="dats"
>
  hello
</h2>
`;

exports[`email input should accept text: 
  <h2 className="valueShow" data-testid="dats">7</h2>
   1`] = `
<h2
  class="valueShow"
  data-testid="dats"
>
  hello
</h2>
`;

exports[`email input should accept text: h1 1`] = `
<h2
  class="valueShow"
  data-testid="dats"
>
  hello
</h2>
`;
